besspin/cyberPhys/specs/state_machine_hacker_kiosk.txt

```diff
@@ -1,95 +1,90 @@
-    ScenarioBaselineHackCriticalSystems: explain hacking critical systems
+Slide7 --> Slide7: A hack is selected
 
-    state "Upload Critical Systtems Hack" as ScenarioBaselineUploadCriticalSystemsHack
-    ScenarioBaselineUploadCriticalSystemsHack: Binary uploaded
-    ScenarioBaselineHackCriticalSystems --> ScenarioBaselineUploadCriticalSystemsHack: Hacker: Choose a hack to upload
+Slide7 --> Slide9: "Next" button pressed
 
-    state "Critical systems hack uploaded" as ScenarioBaselineCriticalSystemsHackUploaded
-    ScenarioBaselineCriticalSystemsHackUploaded: Binary uploaded
-    ScenarioBaselineUploadCriticalSystemsHack --> ScenarioBaselineCriticalSystemsHackUploaded: Send CMD_LED_SEQ(???)
+Slide9: Attempt to hack critical systems
 
-    ScenarioBaselineCriticalSystemsHackUploaded --> ScenarioBaselineHackInfotainment: Return to the hack selection screen
+Slide9 --> Slide10: "Next" button pressed
 
+Slide10 --> Slide10: A hack is selected
 
-    state "Hack Critical Systems" as ScenarioBaselineHackCriticalSystems
-    ScenarioBaselineHackCriticalSystems --> ScenarioBaselineEndOfScenario: Hacker: Clicks on next slide
-    state "Scenario Done" as ScenarioBaselineEndOfScenario
-}
+state "Slide 10 HACK CRITICAL SYSTEMS" as Slide10
+Slide10: When a hack is selected, an appropriate pre-compiled binary is selected
+Slide10: and uploaded using "OtaHack.uploadFile(binary)"
+Slide10: The hack button will color red/green based on the selection
+Slide10: and whether the upload was succesfull
+Slide10: Tx CMD_HACK_ACTIVE(hack-number) (to visualize hacks)
 
-state "Protected Critical Systems Scenario" as ScenarioCritical {
-    state "Introduce SSITH" as IntroSSITH
-    IntroSSITH: Note: new set of FPGAs was selected
+Slide10 --> Slide15: "Next" button pressed
 
-    ScenarioBaselineEndOfScenario --> IntroSSITH: Send CMD_ACTIVE_SCENARIO(CRITICAL_SYSTEMS)\nSend CMD_LED_SEQ(SSITH_CRITICAL)
-}
+Slide15: restart baseline scenario components using CMD_RESTART() messages
+Slide15: Tx CMD_ACTIVE_SCENARIO(SCENARIO_SECURE_ECU) to color CanDisplay
+Slide15: Tx CMD_HACK_ACTIVE(0x0)
 
-KioskTerminate --> [*]: Send CMD_COMPONENT_ERROR(KIOSK|TERMINATED)
+state "Slide 15 SSITH INTRO" as Slide15
+Slide15 --> Slide16: click "Hack Infotainment"
+Slide16: Tx CMD_ACTIVE_SCENARIO(SCENARIO_SECURE_INFOTAINMENT)
 
-'@startuml
-'state "Hacker Kiosk Start up" as HackerKioskUp
-'state "Hacker Kiosk Ready" as HackerKioskReady
-'HackerKioskUp: boot OS
-'HackerKioskUp: connect to the network
-'HackerKioskUp: start Kiosk UI app
-'HackerKioskReady: wait for user
-'[*] --> HackerKioskUp: Hacker Kiosk powered up
-'HackerKioskUp -[#blue]-> HackerKioskReady: Success\nSend CMD_COMPONENT_READY(HACKER_KIOSK)
-'@enduml
+Slide16 --> Slide16: A hack is selected
+Slide16: All hacks fail with an error message (OTA hack not succesfull)
+Slide16: TODO: add a slide where we attempt to hack OTA server
+Slide16: TODO: OTA server crashes on SSITH P2 when a hack is attempted (~2min downtime)
```

Comment:
Is there really a 2 minute downtime when a hack is attempted? Shouldn't the OTA server just crash and then be restarted? Or is that the maximum timeout that would occur before the watchdog restarts it? (presumably we could also just make `service` restart it automatically if it dies...)

Reply:
I measured how long the automatic restart takes a couple of weeks back, and it was around 2 minutes with the auto start (`Restart=on-failure` in the service config). I will re-measure to see how long `systemctl stop/start` takes.